rust/lancedb/src/utils.rs

```diff
@@ -118,14 +118,14 @@ pub(crate) fn default_vector_column(schema: &Schema, dim: Option<i32>) -> Result
         })
         .collect::<Vec<_>>();
     if candidates.is_empty() {
-        Err(Error::Schema {
-            message: "No vector column found to create index".to_string(),
+        Err(Error::InvalidInput {
+            message: "No vector column found to match with the query vector dimension".to_string(),
```

Comment:
can we print the explicit dimension as well?

`format!("No vector column found to match with the query vector dimension: {}", dim.unwrap_or_default())`